utils/subprocess/executor.go

```diff
@@ -192,12 +193,17 @@ func (s *Subprocess) IsOn() bool {
 	return s.isRunning.Load() && s.processMonitoring.IsOn()
 }
 
-// Wait waits for the command to exit and waits for any copying to
-// stdin or copying from stdout or stderr to complete.
-//
-// The command must have been started by Start.
-func (s *Subprocess) Wait() error {
-	return s.command.cmdWrapper.cmd.Wait()
+// Wait waits for the command to stop existing on the system.
+// This allows check to work if the underlying process was stopped.
+func (s *Subprocess) Wait(ctx context.Context) (err error) {
+	var pid int
+	if s.command != nil && s.command.cmdWrapper.cmd != nil && s.command.cmdWrapper.cmd.Process != nil {
+		pid = s.command.cmdWrapper.cmd.Process.Pid
+	} else {
+		return commonerrors.New(commonerrors.ErrCondition, "command not started")
+	}
+
+	return proc.WaitForCompletion(ctx, pid)
```

Comment:
just so that we know behaviour is not broken on windows. Could you have s temporary //FIXME if platform.isWindows{ call the go wait method as it used to be }